compiler/rustc_ty_utils/src/assoc.rs

```diff
@@ -316,11 +313,12 @@ fn impl_associated_item_for_impl_trait_in_trait(
     trait_assoc_def_id: LocalDefId,
     impl_fn_def_id: LocalDefId,
 ) -> LocalDefId {
-    let impl_def_id = tcx.local_parent(impl_fn_def_id);
+    let impl_def_id = tcx.parent(impl_fn_def_id.to_def_id());
```

Comment:
Why? I prefer the local_parent call, then use `to_def_id` for the places that need a non-local def id

Reply:
I did this change because down I needed `impl_def_id` to be `def_id` and ended doing a lot of `to_def_id` but whatever, can change that if you prefer or use a `impl_local_def_id` var. Let me know what you prefer and I'll change it.

Reply:
Prefer keeping it local and turning it into a def_id when needed. Most of these calls should be able to take both `LocalDefId` and `DefId` because they take `impl IntoQueryParam<DefId>`...?